helm/oauth2-proxy/templates/deployment.yaml

```diff
@@ -237,7 +242,7 @@ spec:
         {{- if .Values.extraEnv }}
 {{ tpl (toYaml .Values.extraEnv) . | indent 8 }}
         {{- end }}
-        {{- if .Values.envFrom }}
+        {{- if and (not .Values.envFromExistingSecret) .Values.envFrom }}
         envFrom:
```

Comment:
@pierluigilenoci I refactored the PR a little bit and gave the variable a better name but then I realised we already have an `envFrom` parameter that allows for using a secret as a source like so:

```
envFrom:
  - secretRef:
      name: my-secret
```

in the `values.yaml` without any changes to the chart.

Reply:
Therefore I don't see the benefit of this addition to the helm chart 🤔

Reply:
The original idea of PR has probably been lost over time. 
However, you can find it in this comment:
https://github.com/oauth2-proxy/manifests/pull/141#issuecomment-1514677797

Reply:
Since then it has been replaced with a more generic solution:
https://github.com/oauth2-proxy/manifests/pull/201

Reply:
I would therefore close this PR.